contractsV5/KyberNetwork.sol

```diff
@@ -100,12 +98,14 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
     /// @dev add or deletes a reserve to/from the network.
     /// @param reserve The reserve address.
     function addReserve(address reserve, uint reserveId, bool isFeePaying, address wallet) public onlyOperator returns(bool) {
-        require(reserveIdToAddresses[reserveId].length == 0);
-        require(reserveAddressToId[reserve] == uint(0));
+        require(reserveIdToAddresses[reserveId].length == 0, "reserveId points to existing reserve");
```

Comment:
in general my idea is
when reserve removed, it will be placed in next array cell.
and cell 0 will be zeroed.
so the check should be cell 0 has zero address....

Reply:
The problem is the first instantiation, when there is no array in the first place. Discovered it when I was writing the test code.

Reply:
Edited this part